src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -67,64 +77,86 @@ public BufferedRecords(
   }
 
   public List<SinkRecord> add(SinkRecord record) throws SQLException {
-    final SchemaPair schemaPair = new SchemaPair(
-        record.keySchema(),
-        record.valueSchema()
-    );
+    final List<SinkRecord> flushed = new ArrayList<>();
+
+    boolean schemaChanged = false;
+    if (!Objects.equals(keySchema, record.keySchema())) {
+      keySchema = record.keySchema();
+      schemaChanged = true;
+    }
+    if (isNull(record.valueSchema())) {
```

Comment:
Oddly even when record.value() returns null, record.valueSchema() returns NOT null for me